src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import java.io.IOException;
+
+public class Parser {
+    private Ui ui;
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Constructor to create instance of Parser
+     * which will make sense of user input
+     * Loads the file from data
+     *
+     */
+    public Parser(Ui ui,Storage storage) {
+        this.ui = ui;
+        this.storage = storage;
+        try {
+            this.taskList = this.storage.load();
+        } catch (IOException error) {
+            this.ui.showError(error.getMessage());
+        }
+    }
+
+    /**
+     * Processes the user input and if
+     * it is exiting, will return false
+     * else return true
+     *
+     * @return Boolean of the process
+     */
+    public String process(String cmd) {
+        try {
+            if (cmd.toLowerCase().equals("bye")) {
```

Comment:
Believe you can extract out the first input and lowercase all of it at once instead of repeating the code